src/policy/satisfy.rs

```diff
@@ -0,0 +1,82 @@
+use crate::core::Value;
+use crate::jet::elements::ElementsEnv;
+use crate::policy::ast::Policy;
+use bitcoin_hashes::sha256;
+use elements::schnorr::{KeyPair, XOnlyPublicKey};
+use elements::secp256k1_zkp::Message;
+use std::collections::HashMap;
+
+#[allow(unused)]
+struct Parameters {
+    pub preimages: HashMap<sha256::Hash, [u8; 32]>,
+    pub keys: HashMap<XOnlyPublicKey, KeyPair>,
+    pub env: ElementsEnv,
+}
```

Comment:
In rust-miniscript, we have a satisfier trait in rust-miniscript to provide abstraction over these values. We have a logic where we compare which satisfaction cost is cheaper when given satisfactions. The API then tries to find the witness with minimum cost. 

This I think is searching if there exists a solution and greedily picking the first one possible. Fine for now, but definitely something we should improve.

Reply:
I will try to use the `Satisfier` trait from rust-miniscript. The signatures for Elements transactions confused me, but a custom implementation might work. The taproot stuff will need some working out since we are using a different leaf version (I think).

Reply:
I don't think I can use `miniscript::Satisfier` directly, since it expects a Bitcoin transaction. I implemented a satisfaction cost that is used to decide on the best disjunctive branches. Thresholds are currently not supported because the underlying Simplicity expression (that I wrote) is pretty horrible (it expects witnesses for unused branches; I will fix that in a future PR).

Reply:
Can you use `CovSatisfier` from `elements-miniscirpt`?

Reply:
Oops, that's a struct. I meant to say `Satisfier`: https://docs.rs/elements-miniscript/latest/elements_miniscript/miniscript/satisfy/trait.Satisfier.html